### PR TITLE
Add hex color validations

### PR DESCRIPTION
**Pull request by tolgap**
I've been looking to get into Elm, and this project seems a great effort to be able to reduce the errors in writing CSS in combination with Elm.

The first item in issue #32 , validation checks are asked for hex values. I have implemented these.

Any feedback is very much welcome as I'm still learning!

### CODE REVIEW

File: src/Css.elm [hunk 1]
```diff
@@ -854,21 +855,25 @@ tools which express these as e.g. `#abcdef0`, etc.
...
-  -- TODO emit warnings for invalid hex chars
   let
     value =
       if String.slice 0 1 str == "#" then
         str
       else
         "#" ++ str
+    warnings =
+      if contains (regex "^#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$") value then
+        [ "Hexadecimal color values can only contain three or six characters of '0-9' and/or 'A-F'. " ++ value ++ " is not valid." ]
```
Comment (eeue56) [inline src/Css.elm]: Generally prefer to join strings with like

```
String.join " "
  [ "hello"
  , value
  , "dog"
  ]
```

Comment (tolgap) [inline src/Css.elm]: @eeue56 I wanted to perform this change, but I noticed that all string concatenation in the file happens with `(++)`. Should I change it anyway?

Comment (rtfeldman) [inline src/Css.elm]: current CSS4 draft [supports lengths of 4 and 8](https://drafts.csswg.org/css-color/#hex-notation) (for alpha); I'm inclined to think allowing those is okay. 😄

Comment (rtfeldman) [inline src/Css.elm]: Also, 👍  for `String.join` here

**Comment by tolgap** [comment]
@rtfeldman I performed the necessary changes.

I could squash the commits if necessary, but GitHub has built in squash merging now. Or do you want to keep the commits as is?

**Comment by rtfeldman** [comment]
Love it, thanks @tolgap! 😻

**Pull request merged**